test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -43,6 +43,92 @@ namespace {
 #define GRPC_ARG_SERVER_SELECTOR_KEY "grpc.testing.server_selector_key"
 #define GRPC_ARG_SERVER_SELECTOR_VALUE "grpc.testing.server_selector_value"
 
+void VerifyAttributes(
```

Comment:
Suggest writing these functions as gmock matchers.  Also, instead of passing in the map and having them do the map lookup and then the vector iteration, suggest writing them to match against only one `opentelemetry::sdk::metrics::PointDataAttributes` and calling them in the context of `::testing::ElementsAre()`.  The goal here should be to write the expectations directly in the test using something like this:

```
EXPECT_THAT(data, ::testing::ElementsAre(
    ::testing::Pair(
        kMetricName,
        ::testing::ElementsAre(
            ::testing::AllOf(
                AttributesEq(kLabelKeys, kLabelValues,
                             kOptionalLabelKeys, kOptionalLabelValues),
                CounterResultEq(kCounterResult))))));
```

This will yield much more helpful error messages when the test fails.